src/mxf_helper/RDD36MXFDescriptorHelper.cpp

```diff
@@ -377,18 +367,71 @@ void RDD36MXFDescriptorHelper::UpdateFileDescriptor(RDD36EssenceParser *essence_
+                match = true;
+            }
+            if (match && color_primaries != g_Null_UL) {
+                // Check at least one color primary matches
+                match = false;
+                for (size_t k = 0; k < BMX_ARRAY_SIZE(DEFAULT_PARAM_MATCHES[i].color_primaries); k++) {
+                    if (DEFAULT_PARAM_MATCHES[i].color_primaries[k] == color_primaries) {
+                        match = true;
+                        break;
+                    } else if (DEFAULT_PARAM_MATCHES[i].color_primaries[k] == g_Null_UL) {
+                        break;
+                    }
+                }
+            }
+            if (match && transfer_characteristic != g_Null_UL) {
+                // Check at least one transfer characteristic matches
+                match = false;
+                for (size_t k = 0; k < BMX_ARRAY_SIZE(DEFAULT_PARAM_MATCHES[i].transfer_characteristic); k++) {
+                    if (DEFAULT_PARAM_MATCHES[i].transfer_characteristic[k] == transfer_characteristic) {
+                        match = true;
+                        break;
+                    } else if (DEFAULT_PARAM_MATCHES[i].transfer_characteristic[k] == g_Null_UL) {
+                        break;
+                    }
+                }
+            }
+            if (match && matrix_coefficients != g_Null_UL) {
+                // Check at least one matrix coefficients matches
+                match = false;
+                for (size_t k = 0; k < BMX_ARRAY_SIZE(DEFAULT_PARAM_MATCHES[i].matrix_coefficients); k++) {
+                    if (DEFAULT_PARAM_MATCHES[i].matrix_coefficients[k] == matrix_coefficients) {
+                        match = true;
+                        break;
+                    } else if (DEFAULT_PARAM_MATCHES[i].matrix_coefficients[k] == g_Null_UL) {
+                        break;
+                    }
+                }
+            }
+            if (match) {
+                if (!cdci_descriptor->haveSignalStandard()) {
+                    // Only set a 296M or 274M signal standard if the colorimetry matches BT709
+                    if (DEFAULT_PARAM_MATCHES[i].signal_standard == MXF_SIGNAL_STANDARD_SMPTE296M ||
+                        DEFAULT_PARAM_MATCHES[i].signal_standard == MXF_SIGNAL_STANDARD_SMPTE274M)
+                    {
+                        if ((color_primaries == ITU709_COLOR_PRIM              || color_primaries == g_Null_UL) &&
+                            (transfer_characteristic == ITUR_BT709_TRANSFER_CH || transfer_characteristic == g_Null_UL) &&
+                            (matrix_coefficients == ITUR_BT709_CODING_EQ       || matrix_coefficients == g_Null_UL))
+                        {
+                            cdci_descriptor->setSignalStandard(DEFAULT_PARAM_MATCHES[i].signal_standard);
+                        }
+                        else
+                        {
+                            cdci_descriptor->setSignalStandard(MXF_SIGNAL_STANDARD_NONE);
```

Comment:
Confirmed by merging this file into our code, building bmx libs and using ProRes HD BT-2020 that Signal Standard 0x00 is written 👍🏻